test/client.test.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const assume = require('assume');
+const sinon = require('sinon');
+const fs = require('fs');
+
+const K8Api = require('../');
+
+describe('K8Api', () => {
+  let sandbox
+
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create();
+  })
+
+  afterEach(() => {
+    sandbox.restore();
+  })
+
+  describe('getInClusterConfig', () => {
+    it('should return with in-cluster config', () => {
+      // may doesn't exist (cannot stub undefined)
+      process.env.KUBERNETES_SERVICE_HOST = process.env.KUBERNETES_SERVICE_HOST || ''
+      process.env.KUBERNETES_SERVICE_PORT = process.env.KUBERNETES_SERVICE_PORT || ''
+
+      sandbox.stub(process.env, 'KUBERNETES_SERVICE_HOST', 'myhost')
+      sandbox.stub(process.env, 'KUBERNETES_SERVICE_PORT', 443)
+
+      const fsReadFileSync = sandbox.stub(fs, 'readFileSync')
+
+      fsReadFileSync
+        .withArgs('/var/run/secrets/kubernetes.io/serviceaccount/ca.crt')
+        .returns('my-cert')
+
+      fsReadFileSync
+        .withArgs('/var/run/secrets/kubernetes.io/serviceaccount/token')
+        .returns('my-token')
+
+      fsReadFileSync
+        .withArgs('/var/run/secrets/kubernetes.io/serviceaccount/namespace')
+        .returns('my-namespace')
+
+      const config = K8Api.getInClusterConfig();
+      assume(config).eqls({
```

Comment:
`equals` is for reference based comprasion: https://www.npmjs.com/package/assume#equal-equals-eq-eqs-exactly
`eqls`: https://www.npmjs.com/package/assume#eql-eqls checks deeply